dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/preheat/supplier/OrgUnitValueTypeSupplierTest.java

```diff
@@ -0,0 +1,279 @@
+        when( manager.getByUid( OrganisationUnit.class, List.of( "kKacJUdANDC" ) ) )
+            .thenReturn( List.of( orgUnit ) );
+
+        TrackerImportParams params = params( TrackerIdSchemeParams.builder().build() )
+            .enrollments(
+                List.of( enrollment( numericAttribute(), orgUnitAttribute( "hQKI6KcEu5t", "kKacJUdANDC" ) ) ) )
+            .build();
+
+        supplier.preheatAdd( params, preheat );
+
+        assertContainsOnly( preheat.getAll( OrganisationUnit.class ), orgUnit );
+    }
+
+    @Test
+    void testSupplierAddsOrgUnitReferencedByEventDataElement()
+    {
+        preheat.put( TrackerIdSchemeParam.UID,
+            List.of( numericDataElement( "numeric" ), orgUnitDataElement( "hQKI6KcEu5t" ) ) );
+
+        OrganisationUnit orgUnit = orgUnit( "kKacJUdANDC" );
+        when( manager.getByUid( OrganisationUnit.class, List.of( "kKacJUdANDC" ) ) )
+            .thenReturn( List.of( orgUnit ) );
+
+        TrackerImportParams params = params( TrackerIdSchemeParams.builder().build() )
+            .events( List.of( event( dataValue( "numeric", "2" ), dataValue( "hQKI6KcEu5t", "kKacJUdANDC" ) ) ) )
+            .build();
+
+        supplier.preheatAdd( params, preheat );
+
+        assertContainsOnly( preheat.getAll( OrganisationUnit.class ), orgUnit );
+    }
+
+    @Test
+    void testSupplierDoesNotAddOrgUnitIfEventDataValueValueIsEmpty()
+    {
+        preheat.put( TrackerIdSchemeParam.UID, List.of( orgUnitDataElement( "hQKI6KcEu5t" ) ) );
+
+        TrackerImportParams params = params( TrackerIdSchemeParams.builder().build() )
+            .events( List.of( event( dataValue( "hQKI6KcEu5t", "" ) ) ) )
+            .build();
+
+        supplier.preheatAdd( params, preheat );
+
+        assertEquals( Collections.emptyList(), preheat.getAll( OrganisationUnit.class ) );
+    }
+
+    private TrackedEntityAttribute teaNumeric( String uid )
+    {
+        TrackedEntityAttribute attribute = createTrackedEntityAttribute( 'A' );
+        attribute.setUid( uid );
+        attribute.setValueType( ValueType.NUMBER );
+        return attribute;
+    }
+
+    private TrackedEntityAttribute teaOrgUnit( String uid )
+    {
+        TrackedEntityAttribute attribute = createTrackedEntityAttribute( 'A' );
+        attribute.setUid( uid );
+        attribute.setValueType( ValueType.ORGANISATION_UNIT );
+        return attribute;
+    }
+
+    private OrganisationUnit orgUnit( String uid )
+    {
+        OrganisationUnit orgUnit = createOrganisationUnit( 'A' );
+        orgUnit.setUid( uid );
+        return orgUnit;
+    }
+
+    private TrackerImportParams.TrackerImportParamsBuilder params( TrackerIdSchemeParams idSchemes )
+    {
+        return TrackerImportParams.builder().idSchemes( idSchemes );
+    }
+
+    private Attribute numericAttribute()
+    {
+
+        return Attribute.builder()
+            .attribute( MetadataIdentifier.ofUid( "numeric" ) )
```

Comment:
we should avoid line breaks evenly in code

Reply:
what do you mean by that? You can use the `Add a suggestion` feature to show how you would format the code 😄

Reply:
in general no empty lines at the beginning and end of a method, we can see if we can add that in the formatter?